apis/python/src/tiledbsoma/_exception.py

```diff
@@ -138,8 +138,23 @@ def is_duplicate_group_key_error(e: Union[SOMAError, tiledb.TileDBError]) -> boo
 
     Lifecycle: maturing
     """
-    stre = str(e)
-    if "member already exists in group" in stre:
-        return True
+    return "member already exists in group" in str(e)
 
-    return False
+
+def is_domain_setting_error(e: SOMAError) -> bool:
+    """Given a SOMAError, return whether it attempted to create the ArraySchema
+    but resulted in a
+
+    Lifecycle: maturing
+    """
+    return "Cannot set domain" in str(e)
+
+
+def map_exception_for_create(e: SOMAError, uri: str) -> None:
```

Comment:
If this is intended to be used in error-handling code, it might be better to have this return the error and raise the new one, so that the caller can put `from None` or `from prev_error` on it. Otherwise, instead of getting:

```
AlreadyExistsError: "some_uri" already exists"
... traceback ...
the cause of this error was ...
```

you get

```
SOMAError: ...
While that was being handled, this happened:
AlreadyExistsError: ...
```

so then at the callsite it looks like:

```python
try:
  ...
except SOMAError as se:
  raise map_exception_for_create(se, ...) from se  # or "from None" if you prefer